lib/resty/healthcheck.lua

```diff
@@ -919,8 +960,25 @@ function checker:run_single_check(ip, port, hostname, hostheader)
   end
 
   local req_headers = self.checks.active.headers
-  local headers = table.concat(req_headers, "\r\n")
-  if #headers > 0 then
+  local headers
+  local headers_length = nkeys(req_headers)
+  if headers_length > 0 then
+    if is_array(req_headers) then
+      self:log(WARN, "array headers is deprecated")
+      headers = table.concat(req_headers, "\r\n")
+    else
+      headers = new_tab(0, headers_length)
+      local idx = 0
+      for k, v in pairs(req_headers) do
+          -- `k` is header name and `v` is header value
+          idx = idx + 1
+          headers[idx] = escape_uri(k) .. ":" .. escape_uri(v)
```

Comment:
I don't recall off the top of my head, but it might be worth double-checking if the `typedefs.headers` input that we are expecting will possibly include tables in the header values. For instance, something like this:

```lua
{
  foo = "bar",
  multi = {"baz", "bat"},
}
```

...thus resulting in the following in the request text:

```
foo: bar
multi: baz
multi: bat
```

Actually that might even be the default "shape" of `typedefs.headers` (`map<string, string[]>`), now that I think of it.

Reply:
Thanks! solved I think